src/Compilers/CSharp/Test/Semantic/Semantics/BindingTests.cs

```diff
@@ -3364,6 +3364,26 @@ public static int MathMax3(this int a, int b)
             Assert.False(symbols.Where(s => s.Name == "MathMax3").Any());
         }
 
+        [Fact, WorkItem(30726, "https://github.com/dotnet/roslyn/issues/30726")]
+        public void UsingStaticGenericConstraint()
+        {
+            var code = @"
+using static Test<System.String>;
```

Comment:
Note that the diagnostic we produce will squiggle all of `Test<System.String>` instead of just `System.String`. This is maybe not perfect (why not squiggle the particular type argument that's wrong?) but shouldn't impact usability much.